libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFDirectoryManager.m

```diff
@@ -45,7 +45,7 @@ + (BOOL)ensureDirectoryExists:(NSString*)directory error:(NSError**)error {
     if (![manager fileExistsAtPath:directory]) {
         return [manager createDirectoryAtPath:directory
                   withIntermediateDirectories:YES
-                                   attributes:@{NSFileProtectionKey: NSFileProtectionComplete}
+                                   attributes:@{NSFileProtectionKey: NSFileProtectionCompleteUntilFirstUserAuthentication}
```

Comment:
@khawkins indeed and upgrade step is needed and is missing here, I'll add that tomorrow.